Intermittent random problems in the Windows leg of the build pipeline

See, for example, [these failures](https://github.com/microsoft/InnerEye-DeepLearning/pull/509/checks?check_run_id=3069179840):

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\_pytest\main.py", line 240, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\_pytest\main.py", line 296, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\pluggy\callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\xdist\dsession.py", line 112, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\xdist\dsession.py", line 135, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "c:\miniconda\envs\innereye\lib\site-packages\xdist\dsession.py", line 177, in worker_workerfinished
INTERNALERROR>     assert not crashitem, (crashitem, node)
INTERNALERROR> AssertionError: ('Tests/ML/test_model_training.py::test_valid_model_train[False]', <WorkerController gw0>)
INTERNALERROR> assert not 'Tests/ML/test_model_training.py::test_valid_model_train[False]'
```

[AB#4254](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/4254)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -43,7 +43,8 @@ mounting and running matplotblib on some machines. Re-instantiated a disabled te
 
 ### Removed
 
-- ([#520](https://github.com/microsoft/InnerEye-DeepLearning/pull/520)) Disable glaucoma job from Azure pipeline.
+-  ([#541](https://github.com/microsoft/InnerEye-DeepLearning/issues/541)) Removed Windows test leg from build pipeline.
+-  ([#520](https://github.com/microsoft/InnerEye-DeepLearning/pull/520)) Disable glaucoma job from Azure pipeline.
 
 ### Deprecated
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -43,7 +43,7 @@ mounting and running matplotblib on some machines. Re-instantiated a disabled te
 
 ### Removed
 
--  ([#541](https://github.com/microsoft/InnerEye-DeepLearning/issues/541)) Removed Windows test leg from build pipeline.
+-  ([#542](https://github.com/microsoft/InnerEye-DeepLearning/pull/542)) Removed Windows test leg from build pipeline.
 -  ([#520](https://github.com/microsoft/InnerEye-DeepLearning/pull/520)) Disable glaucoma job from Azure pipeline.
 
 ### Deprecated

--- a/azure-pipelines/build-pr.yml
+++ b/azure-pipelines/build-pr.yml
@@ -21,7 +21,7 @@ jobs:
     pool:
       vmImage: 'windows-2019'
     steps:
-      - template: build.yaml
+      - template: build_windows.yaml
 
   - job: Linux
     pool:

--- a/azure-pipelines/build.yaml
+++ b/azure-pipelines/build.yaml
@@ -1,19 +1,6 @@
 steps:
   - template: checkout.yml
 
-  - task: CredScan@3
-    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
-
-  - task: PostAnalysis@1
-    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
-    displayName: 'Post Analysis'
-    inputs:
-      CredScan: true
-
-  - script: echo %NUMBER_OF_PROCESSORS%
-    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
-    displayName: Print processors
-
   - bash: |
       conda env create --file environment.yml --name InnerEye --quiet
       source activate InnerEye

--- a/azure-pipelines/build_windows.yaml
+++ b/azure-pipelines/build_windows.yaml
@@ -10,6 +10,3 @@ steps:
     inputs:
       CredScan: true
 
-  - script: echo %NUMBER_OF_PROCESSORS%
-    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
-    displayName: Print processors

--- a/azure-pipelines/build_windows.yaml
+++ b/azure-pipelines/build_windows.yaml
@@ -10,3 +10,11 @@ steps:
     inputs:
       CredScan: true
 
+  - task: ComponentGovernanceComponentDetection@0
+    condition: succeeded()
+    inputs:
+      scanType: 'Register'
+      verbosity: 'Normal'
+      alertWarningLevel: 'High'
+      failOnAlert: true
+      failOnStderr: true

--- a/azure-pipelines/build_windows.yaml
+++ b/azure-pipelines/build_windows.yaml
@@ -1,0 +1,14 @@
+  - template: checkout_windows.yml
+
+  - task: CredScan@3
+    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+
+  - task: PostAnalysis@1
+    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+    displayName: 'Post Analysis'
+    inputs:
+      CredScan: true
+
+  - script: echo %NUMBER_OF_PROCESSORS%
+    condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ))
+    displayName: Print processors

--- a/azure-pipelines/build_windows.yaml
+++ b/azure-pipelines/build_windows.yaml
@@ -1,3 +1,4 @@
+steps:
   - template: checkout_windows.yml
 
   - task: CredScan@3

--- a/azure-pipelines/checkout.yml
+++ b/azure-pipelines/checkout.yml
@@ -4,10 +4,7 @@ steps:
     submodules: true
 
   - bash: |
-      if [ $(Agent.OS) = 'Windows_NT' ]
-      then subdir=Scripts
-      else subdir=bin
-      fi
+      subdir=bin
       echo "Adding this directory to PATH: $CONDA/$subdir"
       echo "##vso[task.prependpath]$CONDA/$subdir"
     displayName: Add conda to PATH
@@ -19,7 +16,6 @@ steps:
       conda list
     displayName: Print conda version and initial package list
 
-  # Linux only; not needed for Windows
   - bash: |
       sudo chown -R $USER /usr/share/miniconda
     condition: and(succeeded(), eq( variables['Agent.OS'], 'Linux' ))

--- a/azure-pipelines/checkout_windows.yml
+++ b/azure-pipelines/checkout_windows.yml
@@ -1,0 +1,17 @@
+steps:
+  - checkout: self
+    lfs: true
+    submodules: true
+
+  - bash: |
+      subdir=Scripts
+      echo "Adding this directory to PATH: $CONDA/$subdir"
+      echo "##vso[task.prependpath]$CONDA/$subdir"
+    displayName: Add conda to PATH
+    condition: succeeded()
+
+  - bash: |
+      conda install conda=4.8.3 -y
+      conda --version
+      conda list
+    displayName: Print conda version and initial package list

